### PR TITLE
OSPool: automatically update user prios

### PR DESCRIPTION
**Pull request by rynge**

### CODE REVIEW

File: opensciencegrid/ospool-cm/ospool.cron [hunk 1]
```diff
@@ -0,0 +1,8 @@
+
+# prios
+ */5 * * * *  root  /opt/ospool/update-prios >/dev/null 2>&1
+
+# web
+ */5 * * * *  root  cd /shared/web/overview && ./utils/cron-5m.sh >/dev/null 2>&1
+  0  * * * *  root  cd /shared/web/overview && ./utils/cron-hourly.sh >/dev/null 2>&1
```
Comment (brianhlin) [inline opensciencegrid/ospool-cm/ospool.cron:7]: Seems like an unrelated change that should at the very least go into a separate commit

File: opensciencegrid/ospool-cm/opt/ospool/update-prios [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    new_factor = current_factor
+
+    # ap20/21 migration
+    if re.search("login0[45].osgconnect.net", ad["Name"]):
+        new_factor = 500
+
+    if new_factor != current_factor:
+        user = ad["Name"]
+        print(f"Updating {user} from {current_factor} to {new_factor}")
+        negotiator.setFactor(user, new_factor)
```
Comment (brianhlin) [inline opensciencegrid/ospool-cm/opt/ospool/update-prios:19]: You should pass the negotiator object as an arg instead of relying on the global scope

File: opensciencegrid/ospool-cm/opt/ospool/update-prios [hunk 3]
```diff
@@ -0,0 +1,27 @@
+#!/bin/python3
+
+import re
+import htcondor
+from pprint import pprint
```
Comment (brianhlin) [inline opensciencegrid/ospool-cm/opt/ospool/update-prios:5]: Doesn't look like you're using this

File: opensciencegrid/ospool-cm/opt/ospool/update-prios [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        user = ad["Name"]
+        print(f"Updating {user} from {current_factor} to {new_factor}")
+        negotiator.setFactor(user, new_factor)
+
+
+negotiator = htcondor.Negotiator()
+prios = negotiator.getPriorities()
+
+for ad in negotiator.getPriorities():
+    update_user(ad)
```
Comment (brianhlin) [inline opensciencegrid/ospool-cm/opt/ospool/update-prios:26]: If this throws an exception for any reason, subsequent users won't get get prio updates. Is that what we want?

File: opensciencegrid/ospool-cm/opt/ospool/update-prios [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+import re
+import htcondor
+from pprint import pprint
+
+
+def update_user(ad):
+    current_factor = int(ad["PriorityFactor"])
+    new_factor = current_factor
+
+    # ap20/21 migration
```
Comment (brianhlin) [inline opensciencegrid/ospool-cm/opt/ospool/update-prios:12]: This could use some clarification: what's the goal of setting the factor to 500 for login04/05 users? To my untrained eye, it's an arbitrary number

**Review by brianhlin — APPROVED** [review]
Does `update-prios` have the executable bit set? I can't figure it out from the GH interface.

See comments in-line but once those are resolve this LGTM. Pre-approving

File: opensciencegrid/ospool-cm/ospool.cron (hunk 1 above)
Comment (rynge) [reply opensciencegrid/ospool-cm/ospool.cron]: Removed in new commit.
